### PR TITLE
Feat(FWS-40): Ability to Filter Invoices by Fields

### PR DESCRIPTION
**Pull request by Abdur-Siddiqui**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/FWS-40

Context:
This ticket is about the filter functionality for the invoices table.

Changes
Added search filter feature in invoices
Added filter button in TrainerInvoices and Admin
Affects Personal Trainer and Admin
-Added 4 kinds of testing
- Added localization

UI:
Changes

image

image

### CODE REVIEW

File: fitwsarah_frontend/src/components/AdminPanel/Filter.js [hunk 1]
```diff
@@ -79,13 +92,29 @@ function Filter({src, labels, onInputChange, searchTerm, clearFilters}) {
                                         )}
                                         {src === "feedback" && (
                                         <>
-                                        {/*The names are the same english or french*/}
+                                        {}
```
Comment (Eli293) [inline fitwsarah_frontend/src/components/AdminPanel/Filter.js:95]: [nitpick]: you should probably get rid of useless brackets.

Comment (Abdur-Siddiqui) [reply fitwsarah_frontend/src/components/AdminPanel/Filter.js:95]: no worries thx for ur comment, ill do it right away.

File: src/test/java/com/fitwsarah/fitwsarah/accountsubdomain/businesslayer/InvoiceServiceUniyTest.java [hunk 2]
```diff
@@ -79,17 +85,51 @@ void createInvoice_ShouldReturnInvoice() {
...
+    @Test
+    void getAllInvoices_ShouldFilterAndReturnInvoices() {
+        // Arrange
+        String invoiceId = "inv1";
+        String userId = "user1";
+        String userName = "John";
+        String status = "COMPLETED";
+        String paymentType = "Credit Card";
+
+        Invoices invoice1 = new Invoices(); // Assume Invoices is a class representing your invoice entity
```
Comment (Eli293) [inline src/test/java/com/fitwsarah/fitwsarah/accountsubdomain/businesslayer/InvoiceServiceUniyTest.java:97]: Maybe get rid of chat comments?

**Review by Eli293 — APPROVED** [review]
Looks good

**Review by AlexeiTimbro — APPROVED** [review]
Good job ABDUR keep it up family.

File: src/test/java/com/fitwsarah/fitwsarah/accountsubdomain/businesslayer/InvoiceServiceUniyTest.java (hunk 2 above)
Comment (Abdur-Siddiqui) [reply src/test/java/com/fitwsarah/fitwsarah/accountsubdomain/businesslayer/InvoiceServiceUniyTest.java:97]: thx just did that

**Review by AlexeiTimbro — APPROVED** [review]
Great job ABDUR keep it up.

**Pull request merged**